src/host.cpp

```diff
@@ -216,43 +227,143 @@ namespace rhost {
             return true;
         }
 
-        void get_blob(const message& msg) {
-            assert(!strcmp(msg.name(), "?GetBlob"));
+        void destroy_blob(blobs::blob_id blob_id) {
+            std::lock_guard<std::mutex> lock(blobs_mutex);
+            blobs.erase(blob_id);
+        }
+
+        void destroy_blobs(const message& msg) {
+            assert(!strcmp(msg.name(), "!DestroyBlob"));
+
+            auto json = msg.json();
+
+            std::lock_guard<std::mutex> lock(blobs_mutex);
+            for (auto val : json) {
+                if (!val.is<double>()) {
+                    fatal_error("DestroyBlob: non-numeric blob ID");
+                }
+
+                auto id = static_cast<blobs::blob_id>(val.get<double>());
+                blobs.erase(id);
+            }
+        }
+
+        void get_blob_size(const message& msg) {
+            assert(!strcmp(msg.name(), "?GetBlobSize"));
 
             auto json = msg.json();
             if (!json[0].is<double>()) {
-                fatal_error("GetBlob: non-numeric blob ID");
+                fatal_error("GetBlobSize: non-numeric blob ID");
             }
             auto id = static_cast<blobs::blob_id>(json[0].get<double>());
 
             std::lock_guard<std::mutex> lock(blobs_mutex);
             auto& it = blobs.find(id);
             if (it == blobs.end()) {
-                fatal_error("GetBlob: no blob with ID %llu", id);
+                fatal_error("GetBlobSize: no blob with ID %llu", id);
             }
 
-            respond_to_message(msg, it->second);
+            respond_to_message(msg, static_cast<double>(it->second.size()));
```

Comment:
This will be 64-bit. Can you add the double overflow check here also? Might be time to put it into an inline function in util.h, something like `ensure_fits_double`.